python/cog/server/worker.py

```diff
@@ -480,20 +501,19 @@ async def _aloop(
         task = None
 
         with scope(self._loop_scope()), redirector:
-            while True:
-                e = cast(Envelope, await self._events.recv())
-                if isinstance(e.event, Cancel) and task and self._cancelable:
-                    task.cancel()
-                elif isinstance(e.event, Shutdown):
-                    break
-                elif isinstance(e.event, PredictionInput):
-                    task = asyncio.create_task(
-                        self._apredict(e.tag, e.event.payload, predict, redirector)
-                    )
-                else:
-                    print(f"Got unexpected event: {e.event}", file=sys.stderr)
-            if task:
-                await task
+            async with asyncio.TaskGroup() as tg:
```

Comment:
Consider this an "I'm unfamiliar with the details of asyncio" question: could this be combined?

```py
        async with scope(self._loop_scope()), redirector, asyncio.TaskGroup() as tg:
```

Reply:
Not as it currently stands because the first two don't implement `__aenter__` and `__aexit__`. Perhaps they could, but they don't need to so I think this is fine as-is.